From 377e3e6e56d318da1b1034790df7a597767e1aae Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Thu, 01 Aug 2024 19:29:39 +0300
Subject: [PATCH] [SW-0] gaudi3: sec: align lbw rr config

Recently there are modifications in preboot_rtos RR
configuration. Following are the patches for reference

https://gerrit.habana-labs.com/#/c/439108/
https://gerrit.habana-labs.com/#/c/438251/

This patch align LBW RR configurations with preboot_rtos
implementation.

Change-Id: I8dc24c7e3a33f20670697bbd50981395c5451721
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/gaudi3/common/security.c b/arch/gaudi3/common/security.c
index 5168a54..e7d236c 100644
--- a/arch/gaudi3/common/security.c
+++ b/arch/gaudi3/common/security.c
@@ -905,20 +905,15 @@
 
 	min = HL_CFG_BASE_FULL + mmD0_PSOC_BTL_BASE;
 	max = HL_CFG_BASE_FULL + mmD0_PARC_INT_GEN_BASE + mmINTR_GEN_MSG2WIRE_SH_HD;
-	hl_secure_lbw_rr_common(&data, HL_SEC_LBW_RR_SHORT_2, min, max, rr_sel, true);
+	hl_secure_lbw_rr_common(&data, HL_SEC_LBW_RR_FULL_0, min, max, rr_sel, true);
 
 	min = HL_CFG_BASE_FULL + mmD0_PARC_INT_GEN_BASE + mmINTR_GEN_MSG2WIRE_SW + 4;
-	max = HL_CFG_BASE_FULL + mmD0_SPDMA0_DUP_ENG_BASE - 1;
-	hl_secure_lbw_rr_common(&data, HL_SEC_LBW_RR_SHORT_3, min, max, rr_sel, true);
-
-	min = HL_CFG_BASE_FULL + mmD0_SPDMA0_DUP_MSTR_IF_AXADDR_LBW_BASE;
 	max = HL_CFG_BASE_FULL + mmD0_CPU_CA53_CFG_BASE + D0_CPU_CA53_CFG_MAX_OFFSET - 1;
-	hl_secure_lbw_rr_common(&data, HL_SEC_LBW_RR_SHORT_4, min, max, rr_sel, true);
+	hl_secure_lbw_rr_common(&data, HL_SEC_LBW_RR_FULL_1, min, max, rr_sel, true);
 
 	min = HL_CFG_BASE_FULL + mmD0_CPU_TIMESTAMP_BASE;
 	max = LBW_BASE + LBW_SIZE - 1;
 	hl_secure_lbw_rr_common(&data, HL_SEC_LBW_RR_SHORT_13, min, max, rr_sel, true);
-
 }
 
 /*
